From 1f2528d1f8d332059ed62130a3579c2eb69e4ac0 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Mon, 18 Jun 2018 12:16:22 +0300
Subject: [PATCH 161/706] enetc: TMP: Workaround MAC filtering hash table model
 issue in L2C, TKT381557

Should be reverted when SI ECO for TKT381557 is available.

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 0b9825a347937bb12c0861be3e1e52348801f9df)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index eff72fd1e517..5698914f2733 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -171,11 +171,11 @@ static void enetc_add_mac_addr_ht_filter(struct enetc_mac_filter *filter,
 
 static void enetc_clear_mac_ht_flt(struct enetc_hw *hw, int si_idx, int type)
 {
-	if (type == UC) {
-		enetc_port_wr(hw, ENETC_PSIUMHFR0(si_idx), 0);
+	if (type == UC) { // FIXME: Swap UC with MC low bits, TKT381557
+		enetc_port_wr(hw, ENETC_PSIMMHFR0(si_idx), 0);
 		enetc_port_wr(hw, ENETC_PSIUMHFR1(si_idx), 0);
 	} else { /* MC */
-		enetc_port_wr(hw, ENETC_PSIMMHFR0(si_idx), 0);
+		enetc_port_wr(hw, ENETC_PSIUMHFR0(si_idx), 0);
 		enetc_port_wr(hw, ENETC_PSIMMHFR1(si_idx), 0);
 	}
 }
@@ -183,11 +183,11 @@ static void enetc_clear_mac_ht_flt(struct enetc_hw *hw, int si_idx, int type)
 static void enetc_set_mac_ht_flt(struct enetc_hw *hw, int si_idx, int type,
 				 u32 *hash)
 {
-	if (type == UC) {
-		enetc_port_wr(hw, ENETC_PSIUMHFR0(si_idx), *hash);
+	if (type == UC) { // FIXME: Swap UC with MC low bits, TKT381557
+		enetc_port_wr(hw, ENETC_PSIMMHFR0(si_idx), *hash);
 		enetc_port_wr(hw, ENETC_PSIUMHFR1(si_idx), *(hash + 1));
 	} else { /* MC */
-		enetc_port_wr(hw, ENETC_PSIMMHFR0(si_idx), *hash);
+		enetc_port_wr(hw, ENETC_PSIUMHFR0(si_idx), *hash);
 		enetc_port_wr(hw, ENETC_PSIMMHFR1(si_idx), *(hash + 1));
 	}
 }
-- 
2.17.1

